src/MetaMorpho.sol

```diff
@@ -367,12 +370,23 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
 
                 if (config[id].cap != 0) revert ErrorsLib.InvalidMarketRemovalNonZeroCap(id);
 
-                if (MORPHO.supplyShares(id, address(this)) != 0) {
+                uint256 supplyShares = MORPHO.supplyShares(id, address(this));
+
+                if (supplyShares != 0) {
                     if (config[id].removableAt == 0) revert ErrorsLib.InvalidMarketRemovalNonZeroSupply(id);
 
                     if (block.timestamp < config[id].removableAt) {
                         revert ErrorsLib.InvalidMarketRemovalTimelockNotElapsed(id);
                     }
+
+                    bytes32[] memory slot = new bytes32[](1);
+                    slot[0] = MorphoStorageLib.marketTotalSupplyAssetsAndSharesSlot(id);
+                    bytes32[] memory res = MORPHO.extSloads(slot);
+
+                    uint256 totalSupplyAssets = uint128(uint256(res[0]));
+                    uint256 totalSupplyShares = uint256(res[0] >> 128);
+
+                    lostAssets += supplyShares.toAssetsDown(totalSupplyAssets, totalSupplyShares);
```

Comment:
Not sure the saved gas is worth the low-levelness of these lines (especially in an admin function).

Reply:
What do you think @morpho-org/onchain-team ? I agree that it makes the code less readable just for an edge case (force removing a market with supply)

Reply:
Or we could have it factorized in the external library of blue or in a dedicated library in MetaMorpho 😁
In any case, we're talking about 100 gas

Reply:
then let's use the simplest form no?